app/javascript/controllers/qrscanner_controller.js

```diff
@@ -0,0 +1,51 @@
+import QrScanner from 'qr-scanner';
+
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+
+  initialize() {
+    this.qrScanner = null;
+  }
+
+  _isPositiveInteger(string) {
+    const castedString = Number(string);
+    return !(castedString == NaN || castedString < 0 || !Number.isInteger(castedString))
+  }
+
+  _found(result) {
```

Comment:
It would help future developer if you add a small comment about the structure of the string you're looking for.